src/sdc11073/wsdiscovery/service.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from sdc11073.xml_types import wsd_types
+
+if TYPE_CHECKING:
+    from lxml.etree import QName
+
+
+class Service:
+    """Service objects contain discovery relevant data of a service.
+
+    They are used for publishing a service or as result (list) when searching for services in the network.
+    """
+
+    def __init__(self,
+                 types: list[QName] | None,
+                 scopes: wsd_types.ScopesType | None,
+                 x_addrs: list[str] | None,
+                 epr: str,
+                 instance_id: str,
+                 metadata_version=1):
+        self.types = types
+        if scopes is not None:
+            assert isinstance(scopes, wsd_types.ScopesType)
+        self.scopes = scopes
+        self._x_addrs = x_addrs

```

Comment:
Not really, it can be None because the XAddr node can be omitted in XML. which results in a None value when reading the xml. In order to make using x_addr simpler later on, a list is guaranteed.

Reply:
No, I want to see it in logs.